Products/ZenUtils/metricwriter.py

```diff
@@ -186,6 +186,8 @@ def notify(self, context_uuid, context_id, metric, timestamp, value, thresh_even
                 eventKeyPrefix = [context_id]
             for ev in self._thresholds.check(context_uuid, metric, timestamp, value):
                 parts = eventKeyPrefix[:]
+                if metric not in parts:   # We add the metric name to the eventKey. Some transforms use it
+                    parts.append(metric)
```

Comment:
I don't think this is correct. Look at the old code. https://github.com/zenoss/4.2.5-RPS/blob/master/src/core/Products/ZenCollector/daemon.py#L361  the datapoint name was added only eventKey wasn't present in thresh_event_data.  

I think the path rsplit in the old code is what added the datapoint name.